src/card_operations/card_fetch_encrypt_data.c

```diff
@@ -86,7 +88,38 @@
 /*****************************************************************************
  * STATIC VARIABLES
  *****************************************************************************/
+card_error_type_e card_get_wallet_list(
+    card_operation_data_t card_data,
+    const card_fetch_wallet_list_config_t *configuration,
+    card_fetch_wallet_list_response_t *response) {
+  if (NULL == configuration || NULL == response ||
+      NULL == response->wallet_list) {
+    return CARD_OPERATION_DEFAULT_INVALID;
+  }
+  if (CARD_OPERATION_SUCCESS == card_data.error_type) {
+    card_data.nfc_data.status = nfc_list_all_wallet(response->wallet_list);
+
+    if (card_data.nfc_data.status == SW_NO_ERROR ||
+        card_data.nfc_data.status == SW_RECORD_NOT_FOUND) {
+      if (card_data.nfc_data.status == SW_RECORD_NOT_FOUND ||
+          configuration->operation.buzzer_on_success) {
+        buzzer_start(BUZZER_DURATION);
+      }
+      if (!configuration->operation.skip_card_removal) {
+        wait_for_card_removal();
+      }
+      // break;
+    } else {
+      card_handle_errors(&card_data);
+    }
+  }
 
+  response->card_info.pairing_error = card_data.nfc_data.pairing_error;
+  response->card_info.tapped_card = card_data.nfc_data.tapped_card;
+  response->card_info.recovery_mode = card_data.nfc_data.recovery_mode;
+  response->card_info.status = card_data.nfc_data.status;
+  return card_data.error_type;
+}
```

Comment:
Instead of creating this new function you should modify the existing one in `card_fetch_wallet_list` and It seems you don't actually need the response and configuration parameters, you can use everything from `card_data` and a boolean return value.